### PR TITLE
Support optional sendOptions param for solana.signAndSendTransaction API

### PR DESCRIPTION
**Pull request by yrliou**
Resolves https://github.com/brave/brave-browser/issues/23122

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
`npm run test -- brave_browser_tests` and `npm run test -- brave_unit_tests`

### CODE REVIEW

**Comment by darkdh** [comment]
Please also create a doc PR based on https://github.com/brave/brave-wallet-docs/pull/59 to reflect this change.
Or you can just push changes to the branch.

**Comment by yrliou** [comment]
> Please also create a doc PR based on [brave/brave-wallet-docs#59](https://github.com/brave/brave-wallet-docs/pull/59) to reflect this change. Or you can just push changes to the branch.

Cool, just pushed a commit to the branch.

File: components/brave_wallet/browser/solana_transaction.cc [hunk 1]
```diff
@@ -18,6 +18,118 @@
 
 namespace brave_wallet {
 
+namespace {
+
+constexpr char kSendOptions[] = "send_options";
+constexpr char kMaxRetries[] = "maxRetries";
+constexpr char kPreflightCommitment[] = "preflightCommitment";
+constexpr char kSkipPreflight[] = "skipPreflight";
```
Comment (darkdh) [inline components/brave_wallet/browser/solana_transaction.cc:27]: Can we add a comment saying the reason these three keys are camel style is for easy handling JSON provider request from dApps?

File: components/brave_wallet/browser/solana_transaction.cc [hunk 2]
```diff
@@ -18,6 +18,118 @@
...
+SolanaTransaction::SendOptions::~SendOptions() = default;
+SolanaTransaction::SendOptions::SendOptions(
+    const SolanaTransaction::SendOptions&) = default;
+SolanaTransaction::SendOptions::SendOptions(
+    absl::optional<uint64_t> max_retries_param,
+    absl::optional<std::string> preflight_commitment_param,
+    absl::optional<bool> skip_preflight_param)
+    : max_retries(max_retries_param),
+      preflight_commitment(preflight_commitment_param),
+      skip_preflight(skip_preflight_param) {}
```
Comment (darkdh) [inline components/brave_wallet/browser/solana_transaction.cc:46]: These should be `std::move` not copy

File: components/brave_wallet/browser/solana_transaction.cc (hunk 1 above)
Comment (yrliou) [reply components/brave_wallet/browser/solana_transaction.cc:30]: https://github.com/brave/brave-core/pull/13764/commits/01421e648f43c4347572392089a8063ed95d3a1f#diff-813b4a863ad7d6dd38df6f140065b560949a28b3b73cbb16fbd00c6aa5d8d04bR25-R26

File: components/brave_wallet/browser/solana_transaction.cc (hunk 2 above)
Comment (yrliou) [reply components/brave_wallet/browser/solana_transaction.cc]: https://github.com/brave/brave-core/pull/13764/commits/01421e648f43c4347572392089a8063ed95d3a1f#diff-813b4a863ad7d6dd38df6f140065b560949a28b3b73cbb16fbd00c6aa5d8d04bR46-R48

File: components/brave_wallet/browser/solana_transaction.cc [hunk 3]
```diff
@@ -75,12 +76,24 @@ SolanaTransaction::SendOptions::FromValue(const base::Value& value) {
...
   if (max_retries_string &&
       base::StringToUint64(*max_retries_string, &max_retries)) {
     options.max_retries = max_retries;
   }
+
+  // Type of maxRetries is number when it's from dApp requests.
+  // We cap the maximum to 2^53-1 here for double precision, it's safe here
+  // because it does not make sense for dApps to set maxRetries that large.
+  auto max_retries_number = options_dict->FindDouble(kMaxRetries);
+  if (max_retries_number && *max_retries_number >= 0 &&
```
Comment (darkdh) [inline components/brave_wallet/browser/solana_transaction.cc:92]: This should be an `else if` case from previous if

**Review by darkdh — APPROVED** [review]
++

**Pull request merged**